app/code/Oleksandrr/Chat/Controller/Message/Messages.php

```diff
@@ -19,35 +20,56 @@ class Messages extends \Magento\Framework\App\Action\Action implements
      */
     private $logger;
 
+    /**
+     * @var \Magento\Customer\Model\Session
+     */
+    private $userSession;
+
     /**
      * Save constructor.
      * @param \Magento\Framework\App\Action\Context $context
      * @param \Oleksandrr\Chat\Model\ResourceModel\Message\CollectionFactory $messageCollectionFactory
      * @param \Psr\Log\LoggerInterface $logger
+     * @param \Magento\Customer\Model\Session $userSession
      */
     public function __construct(
         \Magento\Framework\App\Action\Context $context,
         \Oleksandrr\Chat\Model\ResourceModel\Message\CollectionFactory $messageCollectionFactory,
-        \Psr\Log\LoggerInterface $logger
+        \Psr\Log\LoggerInterface $logger,
+        \Magento\Customer\Model\Session $userSession
     ) {
         parent::__construct($context);
         $this->messageCollectionFactory = $messageCollectionFactory;
         $this->logger = $logger;
+        $this->userSession = $userSession;
     }
 
     public function execute()
     {
         $collection = [];
+        $message = '';
 
         try {
-            $messageCollection = $this->messageCollectionFactory->create();
-            $messageCollection
-                ->addFieldToSelect('author_name')
-                ->addFieldToSelect('message')
-                ->addFieldToSelect('created_at')
-                ->setPageSize(10);
-            $collection = $messageCollection->getData();
-            $message = 'Post count: ' . count($collection);
+            if ($this->userSession->isLoggedIn()) {
+                $messageCollection = $this->messageCollectionFactory->create()->addAuthorFilter((int) $this->userSession->getId());
+            } else {
+                if ($chatHash = $this->userSession->getChatHash()) {
+                    $messageCollection = $this->messageCollectionFactory->create()->addChatHashFilter($chatHash);
+                }
+            }
+
+            if (isset($messageCollection)) {
+                /** @var Message $message */
+                foreach ($messageCollection as $message) {
+                    $collection[] = [
+                        'author_name' => $message->getAuthorName(),
+                        'message' => $message->getMessage(),
+                        'created_at' => $message->getCreatedAt(),
+                    ];
+                }
+
+                $message = 'Post count: ' . count($collection);
```

Comment:
@pandaUbuntu `$collection` is \Traversable, but better to use `$collection->count()` instead.

Reply:
in this case it is a native array and it has no method of count
$collection = [];

Reply:
@pandaUbuntu You're right. And in this case naming is not really good ) Please, don't use `collection` for non-objects. Can use `$messages` or `$messageList`. By the way, this is not `Post` anyway